### PR TITLE
SSCS-10517 base location fix

### PR DESCRIPTION
**Pull request by wmarshid**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10517

### Change description ###

- Now using RpcVenueService to get the base location
- All associated refactor and updates
- Some code tidy stuff, sorry couldn't help myself!

**Does this PR introduce a breaking change?**

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/bulkscancore/handlers/CcdCallbackHandler.java [hunk 1]
```diff
@@ -211,15 +215,22 @@ && isNotBlank(callback.getCaseDetails().getCaseData().getAppeal().getBenefitType
...
+                            && courtVenue.getRegionId() != null) {
+                            callback.getCaseDetails().getCaseData()
+                                .setCaseManagementLocation(CaseManagementLocation.builder()
+                                    .baseLocation(rpcEpimsId)
+                                    .region(courtVenue.getRegionId())
+                                    .build());
+                        }
                     }
                 }
             }
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/bulkscancore/handlers/CcdCallbackHandler.java:236]: Nested if's like this makes it hard to understand.
Refactor to reduce the nested's to no more than 2 deep

File: src/test/java/uk/gov/hmcts/reform/sscs/service/RpcVenueServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Address;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appellant;
+import uk.gov.hmcts.reform.sscs.ccd.domain.RegionalProcessingCenter;
+import uk.gov.hmcts.reform.sscs.helper.AppellantPostcodeHelper;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RpcVenueServiceTest {
```
[the remark this answers is not in the document]
Comment (nickwest1) [reply src/test/java/uk/gov/hmcts/reform/sscs/service/RpcVenueServiceTest.java:19]: That's a fair comment, but this PR doesn't decide the behaviour, so it should be a comment as food for thought only - no action needed unless @wmarshid disagrees?

**Review by nickwest1 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/service/RpcVenueServiceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (wmarshid) [reply src/test/java/uk/gov/hmcts/reform/sscs/service/RpcVenueServiceTest.java:19]: Agreed.. the requirement is very washy for this one.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/RpcVenueService.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public class RpcVenueService {
+
+    private final AppellantPostcodeHelper appellantPostcodeHelper;
+    private final RegionalProcessingCenterService regionalProcessingCenterService;
+    private final VenueService venueService;
+
+    public String retrieveRpcEpimsIdForAppellant(Appellant appellant) {
+        String appellantPostcode = appellantPostcodeHelper.resolvePostcode(appellant);
+        RegionalProcessingCenter rpc = regionalProcessingCenterService.getByPostcode(appellantPostcode);
+        return venueService.getEpimsIdForActiveVenueByPostcode(rpc.getPostcode()).orElse(null);
```
[the remark this answers is not in the document]
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/RpcVenueService.java:20]: Similar to testing comment @nickwest1 - the requirements aren't the best for this.

**Review by nickwest1 — APPROVED** [review]

**Review by MohammedHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/AppealPostcodeHelper.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        return Optional.ofNullable(appellant.getAppointee())
+            .map(Appointee::getAddress)
+            .map(Address::getPostcode)
+            .filter(this::isValidPostcode)
+            .orElse(Optional.ofNullable(appellant.getAddress())
+                .map(Address::getPostcode)
+                .filter(this::isValidPostcode)
+                .orElse(StringUtils.EMPTY));
```
Comment (TeaWithLucas) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/AppealPostcodeHelper.java:31]: Should only be using appointee's postcode is isAppointee is Yes

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/AppealPostcodeHelper.java:31]: I don't trust bulk scan enough to have this set. If an appointee is provided in the exception record then I think it's safe to assume that this is the case. There's a related ticket to look into and fix the appointee issues in bulk scan

**Review by TeaWithLucas — APPROVED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Pull request merged**